include/envoy/secret/secret_manager.h

```diff
@@ -18,16 +18,25 @@ class SecretManager {
   virtual ~SecretManager() {}
 
   /**
-   * @param secret a protobuf message of envoy::api::v2::auth::Secret.
-   * @throw an EnvoyException if the secret is invalid or not supported.
+   * @param add a static secret from envoy::api::v2::auth::Secret.
+   * @throw an EnvoyException if the secret is invalid or not supported, or there is duplicate.
    */
-  virtual void addOrUpdateSecret(const envoy::api::v2::auth::Secret& secret) PURE;
+  virtual void addStaticSecret(const envoy::api::v2::auth::Secret& secret) PURE;
 
   /**
-   * @param name a name of the Ssl::TlsCertificateConfig.
-   * @return the TlsCertificate secret. Returns nullptr if the secret is not found.
+   * @param name a name of the static TlsCertificateConfigProvider.
+   * @return the TlsCertificateConfigProviderSharedPtr. Returns nullptr if the static secret is not
+   * found.
    */
-  virtual const Ssl::TlsCertificateConfig* findTlsCertificate(const std::string& name) const PURE;
+  virtual TlsCertificateConfigProviderSharedPtr
+  findStaticTlsCertificateProvider(const std::string& name) const PURE;
+
+  /**
+   * @param tls_certificate the protobuf config of the TLS certificate.
+   * @return a TlsCertificateConfigProviderSharedPtr created from tls_certificate.
+   */
+  virtual TlsCertificateConfigProviderSharedPtr createInlineTlsCertificateProvider(
```

Comment:
It could be static with current implementation since I wanted to keep this PR as a refactor. Though secret_manager might do more in future, (e.g. returning same provider if config is exactly same), and/or track all secrets for dump config like rds does.

Reply:
Sorry what is the different between an inline secret and a static secret? I'm confused.

Reply:
inline secret is the secret directly embedded in the config.  static secrets are specified in the static_resources,  the config only has a name to point to it.

Reply:
My understanding is:  inline is the legacy way of configuring ssl cert.   statis secrets are only added when SDS works started.   By removing inline, it may break existing users.

Reply:
I'm not saying to remove inline, I think that will always be there. My question is why isn't inline just a static secret?

Reply:
@mattklein123 IIUC @PiotrSikora asks why not making the C++ method static, your question is to merge inline secret and static secret.
Static secrets can be only registered via static_resources in bootstrap.proto, looked up by name, owned by secret manager. Inline secrets are in each tls_context, have no name, in future, secret manager may track more than current implementation. They have different lifecycles.

Reply:
Ah I see. OK.